Deploy more smart pointers in Source/WebKit/UIProcess/API

rniwa opened the pull request:
#### b6e85c7bf092e2b0315d9d9e4952cd5de33033e1
<pre>
Deploy more smart pointers in Source/WebKit/UIProcess/API
<a href="https://bugs.webkit.org/show_bug.cgi?id=279388">https://bugs.webkit.org/show_bug.cgi?id=279388</a>

Reviewed by Timothy Hatcher.

Deployed more smart pointers in Source/WebKit/UIProcess/API as warned by the clang static analyzer.

* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.mm:
(-[WKWebExtension manifest]):
(-[WKWebExtension manifestVersion]):
(-[WKWebExtension supportsManifestVersion:]):
(-[WKWebExtension defaultLocale]):
(-[WKWebExtension displayName]):
(-[WKWebExtension displayShortName]):
(-[WKWebExtension displayVersion]):
(-[WKWebExtension displayDescription]):
(-[WKWebExtension displayActionLabel]):
(-[WKWebExtension version]):
(-[WKWebExtension iconForSize:]):
(-[WKWebExtension actionIconForSize:]):
(-[WKWebExtension requestedPermissions]):
(-[WKWebExtension optionalPermissions]):
(-[WKWebExtension requestedPermissionMatchPatterns]):
(-[WKWebExtension optionalPermissionMatchPatterns]):
(-[WKWebExtension allRequestedMatchPatterns]):
(-[WKWebExtension errors]):
(-[WKWebExtension hasBackgroundContent]):
(-[WKWebExtension hasPersistentBackgroundContent]):
(-[WKWebExtension hasInjectedContent]):
(-[WKWebExtension hasOptionsPage]):
(-[WKWebExtension hasOverrideNewTabPage]):
(-[WKWebExtension hasCommands]):
(-[WKWebExtension hasContentModificationRules]):
(-[WKWebExtension _hasServiceWorkerBackgroundContent]):
(-[WKWebExtension _hasModularBackgroundContent]):
(-[WKWebExtension _protectedWebExtension]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionAction.mm:
(-[WKWebExtensionAction webExtensionContext]):
(-[WKWebExtensionAction associatedTab]):
(-[WKWebExtensionAction iconForSize:]):
(-[WKWebExtensionAction label]):
(-[WKWebExtensionAction badgeText]):
(-[WKWebExtensionAction hasUnreadBadgeText]):
(-[WKWebExtensionAction setHasUnreadBadgeText:]):
(-[WKWebExtensionAction inspectionName]):
(-[WKWebExtensionAction setInspectionName:]):
(-[WKWebExtensionAction isEnabled]):
(-[WKWebExtensionAction menuItems]):
(-[WKWebExtensionAction presentsPopup]):
(-[WKWebExtensionAction popupViewController]):
(-[WKWebExtensionAction popupPopover]):
(-[WKWebExtensionAction popupWebView]):
(-[WKWebExtensionAction closePopup]):
(-[WKWebExtensionAction _protectedWebExtensionAction]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionActionInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.mm:
(-[WKWebExtensionCommand debugDescription]):
(-[WKWebExtensionCommand webExtensionContext]):
(-[WKWebExtensionCommand setActivationKey:]):
(-[WKWebExtensionCommand modifierFlags]):
(-[WKWebExtensionCommand setModifierFlags:]):
(-[WKWebExtensionCommand menuItem]):
(-[WKWebExtensionCommand _shortcut]):
(-[WKWebExtensionCommand _userVisibleShortcut]):
(-[WKWebExtensionCommand _matchesEvent:]):
(-[WKWebExtensionCommand _protectedWebExtensionCommand]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommandInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.mm:
(-[WKWebExtensionContext initForExtension:]):
(-[WKWebExtensionContext webExtension]):
(-[WKWebExtensionContext permissionStatusForMatchPattern:inTab:]):
(-[WKWebExtensionContext setPermissionStatus:forMatchPattern:expirationDate:]):
(-[WKWebExtensionContext performCommand:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.mm:
(-[WKWebExtensionController initWithConfiguration:]):
(-[WKWebExtensionController configuration]):
(-[WKWebExtensionController extensionContextForExtension:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfiguration.mm:
(-[WKWebExtensionControllerConfiguration copyWithZone:]):
(-[WKWebExtensionControllerConfiguration identifier]):
(-[WKWebExtensionControllerConfiguration webViewConfiguration]):
(-[WKWebExtensionControllerConfiguration setWebViewConfiguration:]):
(-[WKWebExtensionControllerConfiguration defaultWebsiteDataStore]):
(-[WKWebExtensionControllerConfiguration setDefaultWebsiteDataStore:]):
(-[WKWebExtensionControllerConfiguration _protectedWebExtensionControllerConfiguration]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfigurationInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataRecord.mm:
(-[WKWebExtensionDataRecord containedDataTypes]):
(-[WKWebExtensionDataRecord totalSizeInBytes]):
(-[WKWebExtensionDataRecord sizeInBytesOfTypes:]):
(-[WKWebExtensionDataRecord errors]):
(-[WKWebExtensionDataRecord _protectedWebExtensionDataRecord]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPattern.mm:
(-[WKWebExtensionMatchPattern scheme]):
(-[WKWebExtensionMatchPattern host]):
(-[WKWebExtensionMatchPattern path]):
(-[WKWebExtensionMatchPattern string]):
(-[WKWebExtensionMatchPattern matchesAllHosts]):
(-[WKWebExtensionMatchPattern matchesURL:options:]):
(-[WKWebExtensionMatchPattern matchesPattern:options:]):
(-[WKWebExtensionMatchPattern _protectedWebExtensionMatchPattern]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPatternInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.mm:
(-[WKWebExtensionMessagePort isDisconnected]):
(-[WKWebExtensionMessagePort sendMessage:completionHandler:]):
(-[WKWebExtensionMessagePort disconnectWithError:]):
(-[WKWebExtensionMessagePort _protectedWebExtensionMessagePort]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebPushDaemonConnection.mm:
(-[_WKWebPushDaemonConnection getPushPermissionStateForOrigin:completionHandler:]):
(-[_WKWebPushDaemonConnection requestPushPermissionForOrigin:completionHandler:]):
(-[_WKWebPushDaemonConnection setAppBadge:origin:]):
(-[_WKWebPushDaemonConnection subscribeToPushServiceForScope:applicationServerKey:completionHandler:]):
(-[_WKWebPushDaemonConnection unsubscribeFromPushServiceForScope:completionHandler:]):
(-[_WKWebPushDaemonConnection getSubscriptionForScope:completionHandler:]):
(-[_WKWebPushDaemonConnection getNextPendingPushMessage:]):
(-[_WKWebPushDaemonConnection _protectedConnection]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebPushMessage.mm:
(-[_WKWebPushMessage data]):
(-[_WKWebPushMessage _protectedMessage]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebPushSubscriptionData.mm:
(-[_WKWebPushSubscriptionData endpoint]):
(-[_WKWebPushSubscriptionData applicationServerKey]):
(-[_WKWebPushSubscriptionData authenticationSecret]):
(-[_WKWebPushSubscriptionData ecdhPublicKey]):
(-[_WKWebPushSubscriptionData _protectedData]):
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
(WebKit::WebExtensionContext::protectedExtension const):
* Source/WebKit/UIProcess/Extensions/WebExtensionController.h:
(WebKit::WebExtensionController::protectedConfiguration const):

Canonical link: <a href="https://commits.webkit.org/283385@main">https://commits.webkit.org/283385@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/58e5695cb3be34a41402571de1c9b9612b62473e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66145 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45518 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18764 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70177 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16755 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53317 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17036 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/53079 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11663 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69212 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41993 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57263 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33716 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38664 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15631 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/59259 "Build is in progress. Recent messages:") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/60559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14987 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71879 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10100 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/14392 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/60399 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10132 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57326 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60690 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8339 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1974 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10012 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/41326 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42402 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43585 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42146 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.mm:
```diff
@@ -176,137 +176,137 @@ - (instancetype)_initWithResources:(NSDictionary<NSString *, id> *)resources
 - (NSDictionary<NSString *, id> *)manifest
 {
-    return _webExtension->manifest();
+    return [self _protectedWebExtension]->manifest();
```
All of these calls should be `self._protectedWebExtension`.

rniwa replied:
Fixed.

xeenon commented on Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommandInternal.h:
```diff
@@ -42,6 +42,7 @@ template<> struct WrapperTraits<WebExtensionCommand> {
 }
 
 @property (nonatomic, readonly) WebKit::WebExtensionCommand& _webExtensionCommand;
+-(Ref<WebKit::WebExtensionCommand>) _protectedWebExtensionCommand;
```
```suggestion
- (Ref<WebKit::WebExtensionCommand>)_protectedWebExtensionCommand;
```

xeenon commented on Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.mm:
```diff
@@ -93,7 +93,7 @@ - (WKWebExtensionContext *)extensionContextForExtension:(WKWebExtension *)extens
     NSParameterAssert([extension isKindOfClass:WKWebExtension.class]);
 
-    if (auto extensionContext = _webExtensionController->extensionContext(extension._webExtension))
+    if (auto extensionContext = _webExtensionController->extensionContext([extension _protectedWebExtension]))
```
```suggestion
    if (auto extensionContext = _webExtensionController->extensionContext(extension._protectedWebExtension))
```

rniwa replied:
Fixed.

xeenon commented on Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.mm:
```diff
@@ -65,14 +65,14 @@ - (instancetype)initWithConfiguration:(WKWebExtensionControllerConfiguration *)c
         return nil;
 
-    API::Object::constructInWrapper<WebKit::WebExtensionController>(self, configuration._webExtensionControllerConfiguration.copy());
+    API::Object::constructInWrapper<WebKit::WebExtensionController>(self, [configuration _protectedWebExtensionControllerConfiguration]->copy());
```
```suggestion
    API::Object::constructInWrapper<WebKit::WebExtensionController>(self, configuration._protectedWebExtensionControllerConfiguration->copy());
```

rniwa replied:
Fixed.

rniwa committed:
Deploy more smart pointers in Source/WebKit/UIProcess/API
https://bugs.webkit.org/show_bug.cgi?id=279388

Reviewed by Timothy Hatcher.

Deployed more smart pointers in Source/WebKit/UIProcess/API as warned by the clang static analyzer.

* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.mm:
(-[WKWebExtension manifest]):
(-[WKWebExtension manifestVersion]):
(-[WKWebExtension supportsManifestVersion:]):
(-[WKWebExtension defaultLocale]):
(-[WKWebExtension displayName]):
(-[WKWebExtension displayShortName]):
(-[WKWebExtension displayVersion]):
(-[WKWebExtension displayDescription]):
(-[WKWebExtension displayActionLabel]):
(-[WKWebExtension version]):
(-[WKWebExtension iconForSize:]):
(-[WKWebExtension actionIconForSize:]):
(-[WKWebExtension requestedPermissions]):
(-[WKWebExtension optionalPermissions]):
(-[WKWebExtension requestedPermissionMatchPatterns]):
(-[WKWebExtension optionalPermissionMatchPatterns]):
(-[WKWebExtension allRequestedMatchPatterns]):
(-[WKWebExtension errors]):
(-[WKWebExtension hasBackgroundContent]):
(-[WKWebExtension hasPersistentBackgroundContent]):
(-[WKWebExtension hasInjectedContent]):
(-[WKWebExtension hasOptionsPage]):
(-[WKWebExtension hasOverrideNewTabPage]):
(-[WKWebExtension hasCommands]):
(-[WKWebExtension hasContentModificationRules]):
(-[WKWebExtension _hasServiceWorkerBackgroundContent]):
(-[WKWebExtension _hasModularBackgroundContent]):
(-[WKWebExtension _protectedWebExtension]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionAction.mm:
(-[WKWebExtensionAction webExtensionContext]):
(-[WKWebExtensionAction associatedTab]):
(-[WKWebExtensionAction iconForSize:]):
(-[WKWebExtensionAction label]):
(-[WKWebExtensionAction badgeText]):
(-[WKWebExtensionAction hasUnreadBadgeText]):
(-[WKWebExtensionAction setHasUnreadBadgeText:]):
(-[WKWebExtensionAction inspectionName]):
(-[WKWebExtensionAction setInspectionName:]):
(-[WKWebExtensionAction isEnabled]):
(-[WKWebExtensionAction menuItems]):
(-[WKWebExtensionAction presentsPopup]):
(-[WKWebExtensionAction popupViewController]):
(-[WKWebExtensionAction popupPopover]):
(-[WKWebExtensionAction popupWebView]):
(-[WKWebExtensionAction closePopup]):
(-[WKWebExtensionAction _protectedWebExtensionAction]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionActionInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.mm:
(-[WKWebExtensionCommand debugDescription]):
(-[WKWebExtensionCommand webExtensionContext]):
(-[WKWebExtensionCommand setActivationKey:]):
(-[WKWebExtensionCommand modifierFlags]):
(-[WKWebExtensionCommand setModifierFlags:]):
(-[WKWebExtensionCommand menuItem]):
(-[WKWebExtensionCommand _shortcut]):
(-[WKWebExtensionCommand _userVisibleShortcut]):
(-[WKWebExtensionCommand _matchesEvent:]):
(-[WKWebExtensionCommand _protectedWebExtensionCommand]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommandInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.mm:
(-[WKWebExtensionContext initForExtension:]):
(-[WKWebExtensionContext webExtension]):
(-[WKWebExtensionContext permissionStatusForMatchPattern:inTab:]):
(-[WKWebExtensionContext setPermissionStatus:forMatchPattern:expirationDate:]):
(-[WKWebExtensionContext performCommand:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.mm:
(-[WKWebExtensionController initWithConfiguration:]):
(-[WKWebExtensionController configuration]):
(-[WKWebExtensionController extensionContextForExtension:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfiguration.mm:
(-[WKWebExtensionControllerConfiguration copyWithZone:]):
(-[WKWebExtensionControllerConfiguration identifier]):
(-[WKWebExtensionControllerConfiguration webViewConfiguration]):
(-[WKWebExtensionControllerConfiguration setWebViewConfiguration:]):
(-[WKWebExtensionControllerConfiguration defaultWebsiteDataStore]):
(-[WKWebExtensionControllerConfiguration setDefaultWebsiteDataStore:]):
(-[WKWebExtensionControllerConfiguration _protectedWebExtensionControllerConfiguration]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfigurationInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionDataRecord.mm:
(-[WKWebExtensionDataRecord containedDataTypes]):
(-[WKWebExtensionDataRecord totalSizeInBytes]):
(-[WKWebExtensionDataRecord sizeInBytesOfTypes:]):
(-[WKWebExtensionDataRecord errors]):
(-[WKWebExtensionDataRecord _protectedWebExtensionDataRecord]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPattern.mm:
(-[WKWebExtensionMatchPattern scheme]):
(-[WKWebExtensionMatchPattern host]):
(-[WKWebExtensionMatchPattern path]):
(-[WKWebExtensionMatchPattern string]):
(-[WKWebExtensionMatchPattern matchesAllHosts]):
(-[WKWebExtensionMatchPattern matchesURL:options:]):
(-[WKWebExtensionMatchPattern matchesPattern:options:]):
(-[WKWebExtensionMatchPattern _protectedWebExtensionMatchPattern]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMatchPatternInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.mm:
(-[WKWebExtensionMessagePort isDisconnected]):
(-[WKWebExtensionMessagePort sendMessage:completionHandler:]):
(-[WKWebExtensionMessagePort disconnectWithError:]):
(-[WKWebExtensionMessagePort _protectedWebExtensionMessagePort]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebPushDaemonConnection.mm:
(-[_WKWebPushDaemonConnection getPushPermissionStateForOrigin:completionHandler:]):
(-[_WKWebPushDaemonConnection requestPushPermissionForOrigin:completionHandler:]):
(-[_WKWebPushDaemonConnection setAppBadge:origin:]):
(-[_WKWebPushDaemonConnection subscribeToPushServiceForScope:applicationServerKey:completionHandler:]):
(-[_WKWebPushDaemonConnection unsubscribeFromPushServiceForScope:completionHandler:]):
(-[_WKWebPushDaemonConnection getSubscriptionForScope:completionHandler:]):
(-[_WKWebPushDaemonConnection getNextPendingPushMessage:]):
(-[_WKWebPushDaemonConnection _protectedConnection]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebPushMessage.mm:
(-[_WKWebPushMessage data]):
(-[_WKWebPushMessage _protectedMessage]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebPushSubscriptionData.mm:
(-[_WKWebPushSubscriptionData endpoint]):
(-[_WKWebPushSubscriptionData applicationServerKey]):
(-[_WKWebPushSubscriptionData authenticationSecret]):
(-[_WKWebPushSubscriptionData ecdhPublicKey]):
(-[_WKWebPushSubscriptionData _protectedData]):
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
(WebKit::WebExtensionContext::protectedExtension const):
* Source/WebKit/UIProcess/Extensions/WebExtensionController.h:
(WebKit::WebExtensionController::protectedConfiguration const):

Canonical link: https://commits.webkit.org/283385@main
